src/client/models/tankShot.js

```diff
@@ -0,0 +1,869 @@
+            pointsRebuild[pointsRebuild.length - 1].push(numberOfLast);
+
+            return pointsRebuild;
+        };
+
+        const findIntersectionCoordinates = (x1, y1, x2, y2, cX, cY, r) => {
+            /* x1, y1 and x2, y2 - are coordinates of line-segment on canvas
+             * cX, cY and r - are coordinates of center of damage and a radius */
+
+            /* using line equation (y = m*x + k) */
+            let m = ( (y2 - y1) / (x2 - x1) );
+            let k = (y1 - m * x1);
+
+            /* using circle equation (a*x^2 + b*x + c = 0) */
+            let a = (Math.pow(m, 2) + 1);
+            let b = 2 * (m * k - m * cY - cX);
+            let c = ( Math.pow(cY, 2) - Math.pow(r, 2) + Math.pow(cX, 2) - 2 * k * cY + Math.pow(k, 2) );
+
+            let xPlus = Math.round( ( -b + ( Math.sqrt( (Math.pow(b, 2)) -4 * a * c ) ) ) / (2 * a) );
+            let xMinus = Math.round( ( -b - ( Math.sqrt( (Math.pow(b, 2)) -4 * a * c ) ) ) / (2 * a) );
+
+            /* using line equation again to calculate two variants of y */
+            let yPlus = Math.round(m * xPlus + k);
+            let yMinus = Math.round(m * xMinus + k);
+
+            let point1 = [xPlus, yPlus];
+            let point2 = [xMinus, yMinus];
+
+            return [point1, point2];
+        };
+
+        const findInitialAngle = (x, y, cx, cy) => {
+            return Math.atan2((y - cy), (x - cx));
+        };
+
+        const rotateFixed = (cx, cy, r, theta) => {
+            let px2,
+                py2;
+
+            px2 = Math.round( cx + (r * Math.cos(theta)) );
+            py2 = Math.round( cy + (r * Math.sin(theta)) );
+
+            return [px2, py2];
+        };
+
+        const findPointOnSegment = (array, segmentX, segmentY) => {
+            /*defines point which coordinates lays on the line-segment of canvas*/
+            let x1;
+            let y1;
+            let x2;
+            let y2;
+            let foundPoint;
+            let point1;
+            let point2;
+
+            for (let i = 1; i < array.length; i++) {
+                x1 = array[i - 1][0];
+                y1 = array[i - 1][1];
+                x2 = array[i][0];
+                y2 = array[i][1];
+
+                foundPoint = calculateLineEquation(x1, y1, x2, y2, segmentX, segmentY);
+
+                if ( ((y1 <= foundPoint) && (foundPoint <= y2)) || ((y2 <= foundPoint) && (foundPoint <= y1)) ) {
+                    point1 = [x1, y1, (i - 1)];
+                    point2 = [x2, y2, i];
+                    return [point1, point2];
+                }
+            }
+            return null;
+        };
+
+        const calculateLineEquation = (x1, y1, x2, y2, segmentX, segmentY) => {
+            /*defines point which coordinates lays on the line of segment*/
+            let y = Math.round( ( (segmentX - x1) * (y2 - y1) ) / (x2 - x1) + y1 );
+
+            if ( (y - 5) <= segmentY && segmentY <= (y + 5) ) {
+                return y;
+            }
```

Comment:
I was thinking about it but in my opinion - there is not way that **y** could not be found at all.
Or it does not matter and should be returned always in all cases?